### PR TITLE
GEOMESA-1356 The Native API's query-builder should allow temporal bou…

### PR DESCRIPTION
**Pull request by cne1x**
…nds to be (in|ex)clusive

Added a new builder function and corresponding unit tests.

Signed-off-by: Chris Eichelberger <cne1x@ccri.com>

### CODE REVIEW

File: geomesa-native-api/src/test/java/org/locationtech/geomesa/api/GeoMesaQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package org.locationtech.geomesa.api;
```
Comment (jnh5y) [inline geomesa-native-api/src/test/java/org/locationtech/geomesa/api/GeoMesaQueryTest.java]: Headers yo!  (This'll cause the mvn license:check in Travis CI to barf.)

**Comment by elahrvivaz** [comment]
looks good to me

File: geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaQuery.java [hunk 2]
```diff
@@ -45,11 +46,62 @@ public GeoMesaQueryBuilder within(double lx, double ly, double ux, double uy) {
...
+
             return this;
         }
 
+        /**
+         * Converts the current start- and end-point into a valid temporal filter.
+         *
+         * @return the net temporal filter that results from the current temporal bounds
+         */
+        public Filter getTemporalFilter() {
```
Comment (jnh5y) [inline geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaQuery.java]: I like this implementation.

Given it, it looks like we could offer 'after' and 'before', no?

Also, those could be combined.  Builder.after(start).before(end).build would return a between.

**Comment by jnh5y** [comment]
This is good to go once the headers are added.  I'd really like to see 'before' and 'after' added.  

Repeated calls to before/after can just reset the value.

**Comment by jnh5y** [comment]
Looks good to me.  As soon as Travis says 'Green', let's merge it.

**Pull request merged**